eng/scripts/Armstrong-Validation.ps1

```diff
@@ -0,0 +1,200 @@
+
+  $changedSwaggerFiles = $changedFiles.Where({ 
+      $_.EndsWith("main.tf")
+    })
+    
+  return $changedSwaggerFiles
+}
+
+$script:armstrongInstalled = $false
+function Ensure-Armstrong-Installed {
+  if ($script:armstrongInstalled) {
+    # If already checked once in this script, don't log anything further
+    return;
+  }
+
+  $script:armstrongInstalled = $true
+
+  if ($PSVersionTable.OS -like "*Windows*") {
+    LogInfo "Skipping Armstrong install on Windows"
+    return;
+  }
+
+  # Armstrong will be installed under $HOME/go/bin
+  $armstrongPath = Join-Path -Path $HOME -ChildPath "go/bin"
+  $env:PATH += ":$armstrongPath"
+
+  # install golang
+  if (!(Get-Command "go" -ErrorAction SilentlyContinue)) {
+    LogInfo "Installing Go..."
+    sudo apt-get update
+    sudo apt-get install -y golang-go
+  }
+  else {
+    LogInfo "Go is already installed."
+  }
+
+
+  # install armstrong
+  if (!(Get-Command "armstrong" -ErrorAction SilentlyContinue)) {
+    LogInfo "Installing Armstrong..."
+    go install github.com/azure/armstrong@latest
+  }
+  else {
+    LogInfo "Armstrong is already installed."
+  }
+
+}
+
+function Validate-Terraform-Error($repoPath, $filePath) {
+  $fileDirectory = (Split-Path -Parent $filePath)
+  # delete existing old error reports
+  Remove-Item -Path "${fileDirectory}/armstrong_credscan_*" -Recurse -Force *> $null
+
+  # run armstrong credscan
+  $specPath = Join-Path -Path $repoPath -ChildPath "specification"
+  LogInfo "armstrong credscan -working-dir $fileDirectory -swagger-repo $specPath"
+  armstrong credscan -working-dir $fileDirectory -swagger-repo $specPath
+
+  $result = @()
+  # error reports are stored in a directory named armstrong_credscan_<timestamp>
+  Get-ChildItem -Path $fileDirectory -Directory -Filter "armstrong_credscan_*" | ForEach-Object {
+    $errorJsonPath = Join-Path -Path $_.FullName -ChildPath "errors.json"
+    if (Test-Path -Path $errorJsonPath) {
+      Get-Content -Path $errorJsonPath -Raw | ConvertFrom-Json | ForEach-Object {
+        $result += "$_"
+      }
+    }
+  }
+
+  return $result
+}
+
+$repoPath = Resolve-Path "$PSScriptRoot/../.."
+
+$terraformErrors = @()
+
+$filesToCheck = (Get-ChangedTerraformFiles (Get-ChangedFiles $BaseCommitish $TargetCommitish))
+
+if (!$filesToCheck) {
+  LogInfo "No Terraform files found to check"
+}
+else {
+  foreach ($file in $filesToCheck) {
+    LogInfo "Checking $file"
+
+    $fullPath = (Join-Path $repoPath $file)
+
+    $suppression = Get-Suppression $fullPath
+    if ($suppression) {
+      $reason = $suppression["reason"] ?? "<no reason specified>"
+
+      LogInfo "  Suppressed: $reason"
+      # Skip further checks, to avoid potential errors on files already suppressed
+      continue
+    }
+
+    try {
+      Ensure-Armstrong-Installed
+      LogInfo "  Validating errors from Terraform file: $fullPath"
+      $terraformErrors += (Validate-Terraform-Error $repoPath $fullPath)
```

Comment:
Is it possible for a directory to contain multiple files ending with `main.tf`, which would cause this check to do redundant work (running same command in same directory multiple times)?  Or is the assumption there is at most one file ending with `main.tf` per folder?

Reply:
That makes sense. I used `EndsWith("/main.tf")` to make sure the file name is `main.tf` in the new PR.